XenModel/XenAPI-Extensions/VM.cs

```diff
@@ -426,6 +426,27 @@ public bool CanHaveGpu()
             }
         }
 
+        public bool HasSriovRecommendation()
+        {
+            XmlDocument xd = GetRecommendations();
+
+            if (xd == null)
+                return true;
+
+            try
+            {
+                XmlNode xn = xd.SelectSingleNode(@"restrictions/restriction[@field='allow_network_sriov']");
+                if (xn == null || xn.Attributes == null)
+                    return false;
+
+                return true;

```

Comment:
Why do we always return true here? Shouldn't we read the value of this field?

Reply:
Because if the vm has  "allow_network_sriov", the value will be true. If the vm isn't sriov allowed, it won't have the  "allow_network_sriov" in OVF.